var/spack/repos/builtin/packages/libmng/package.py

```diff
@@ -25,11 +28,9 @@ def patch(self):
         filter_file(r'^(\#include \<jpeglib\.h\>)',
                     '#include<stdio.h>\n\\1', 'libmng_types.h')
 
-    @run_before('configure')
-    def clean_configure_directory(self):
-        """Without this, configure crashes with:
+    def cmake_args(self):
+        args = [('WITH_LCMS2', True),
+                ('WITH_LCMS1', False)]
 
-            configure: error: source directory already configured;
-            run "make distclean" there first
-        """
-        make('distclean')
+        return ['-D{0}:BOOL={1}'.format(key, 'ON' if val else 'OFF')
+                for (key, val) in args]
```

Comment:
This seems more complex than it needs to be...

Reply:
Yes... when I first put it in, I had another option or two, and decided I'd format it for easy replacement with
```python
return [self.define(*kv) for kv in args]
```
pending  https://github.com/spack/spack/pull/14376 .

I can replace with an explicit set of arguments no problem.